mlflow/protos/service.proto

```diff
@@ -682,9 +682,18 @@ message ListExperiments {
   // If unspecified, return only active experiments.
   optional ViewType view_type = 1;
 
+  // Maximum number of registered models desired. Max threshold is 50K.
+  optional int64 max_results = 2 [default = 50000];
```

Comment:
how did we arrive at 50k? also nit copy paste in the comment

is this max_results (eg top which only makes sense combined with order_by) or page_size?

Reply:
50k was the analogous limit for registered models but there's absolutely no reason it needs to stay there.

`max_results` is page size - `max_results` is used the same way elsewhere in the in user-facing APIs so I'm sure how much of a choice we have

Reply:
I guess a related design decision is whether we paginate within the `MlflowClient.list_experiments` API, or have it be a thin wrapper over the REST API that returns a PagedList.

I guess for parity with existing APIs we should do the latter (which is what this PR currently does), but seems like if we implemented pagination within `MlflowClient.list_experiments` itself, we could have a separate default for in the REST API and in the client, e.g. have `MlflowClient.list_experiments(max_results=10000)`  but impose a stricter default at the REST API level

Reply:
I've removed the 50k limit from protos. When we have consensus on good limits, I'll make sure they're implemented through the other pieces of the PR.

Reply:
@smurching it looks like removing the default totally leads to problems in the Java client, at least - without something supplied in the protos, it defaults to 0 and pulls no experiments at all. I think having some absurdly high limit like 50k may be necessary, where a more thoughtful limit is then applied in other layers.

Reply:
yeah, just confirmed that if I add the default back the Java and R test suites are passing again.

Reply:
@wamartin-aml could we set the default here to 10,000? Just because it feels a little strange to be able to specify a higher `max_results` by passing nothing at all (50,000) than if you explicitly pass `max_results` (we limit to 10,000)